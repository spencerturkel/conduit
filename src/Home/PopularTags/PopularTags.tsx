import * as React from 'react';
import {StatelessComponent} from 'react';

export interface PopularTagsProps {
    tags: Array<{name: string; link: string}>;
}

const PopularTags: StatelessComponent<PopularTagsProps> = ({tags}: PopularTagsProps) => (
    <div className="sidebar">
        <p>Popular Tags</p>

        <div className="tag-list">
            {tags.map(({name, link}) => (
                <a href={link} className="tag-pill tag-default">
                    {name}
                </a>
            ))}
        </div>
    </div>
);

export default PopularTags;
