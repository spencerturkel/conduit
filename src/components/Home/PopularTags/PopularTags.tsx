import * as React from 'react';
import {StatelessComponent} from 'react';

import setDisplayName from '../../set-display-name';

export interface PopularTagsProps {
    onTagClicked: (tag: string) => void;
    tags: string[];
}

const PopularTags = (): StatelessComponent<PopularTagsProps> => {
    const sfc = ({onTagClicked, tags}: PopularTagsProps) => (
        <div className="sidebar">
            <p>Popular Tags</p>

            <div className="tag-list">
                {tags.map(name => (
                    <button
                        type="button"
                        key={name}
                        onClick={() => onTagClicked(name)}
                        className="tag-pill tag-default">
                        {name}
                    </button>
                ))}
            </div>
        </div>
    );

    setDisplayName(sfc, PopularTags);

    return sfc;
};

export default PopularTags;
