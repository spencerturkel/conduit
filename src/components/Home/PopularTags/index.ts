export {default} from './PopularTags';
export * from './PopularTags';
