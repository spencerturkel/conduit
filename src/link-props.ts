import * as React from 'react';

export interface LinkProps {
    children?: React.ReactNode;
    className: string;
    to: string;
}
