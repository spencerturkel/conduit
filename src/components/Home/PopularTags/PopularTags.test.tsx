import {mount, ReactWrapper} from 'enzyme';
import * as React from 'react';
import PopularTags from './PopularTags';

import {LinkProps} from '../../link-props';

let sut: ReactWrapper;

const TestLink = ({to: href, ...rest}: LinkProps) => <a {...{href}} {...rest} />;

const TestPopularTags = PopularTags(TestLink);
const testTags = [{name: 'programming', link: '/programming'}, {name: 'reactjs', link: '/reactjs'}];

beforeEach(() => {
    sut = mount(<TestPopularTags tags={testTags} />);
});

it('should match snapshot', () => {
    expect(sut).toMatchSnapshot();
});
