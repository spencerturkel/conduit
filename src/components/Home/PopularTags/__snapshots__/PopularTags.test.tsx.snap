// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <PopularTags(TestLink)
    tags={
        Array [
            Object {
              "link": "/programming",
              "name": "programming",
            },
            Object {
              "link": "/reactjs",
              "name": "reactjs",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "tags": Array [
        Object {
          "link": "/programming",
          "name": "programming",
        },
        Object {
          "link": "/reactjs",
          "name": "reactjs",
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="sidebar"
>
        <p>
                Popular Tags
        </p>
        <div
                class="tag-list"
        >
                <a
                        href="/programming"
                        class="tag-pill tag-default"
                >
                        programming
                </a>
                <a
                        href="/reactjs"
                        class="tag-pill tag-default"
                >
                        reactjs
                </a>
        </div>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Popular Tags
</p>,
          <div
            className="tag-list"
>
            <TestLink
                        className="tag-pill tag-default"
                        to="/programming"
            >
                        programming
            </TestLink>
            <TestLink
                        className="tag-pill tag-default"
                        to="/reactjs"
            >
                        reactjs
            </TestLink>
</div>,
        ],
        "className": "sidebar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <p>
            Popular Tags
</p>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Popular Tags",
          },
          "ref": null,
          "rendered": Array [
            "Popular Tags",
          ],
          "type": "p",
        },
        Object {
          "instance": <div
            class="tag-list"
>
            <a
                        href="/programming"
                        class="tag-pill tag-default"
            >
                        programming
            </a>
            <a
                        href="/reactjs"
                        class="tag-pill tag-default"
            >
                        reactjs
            </a>
</div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TestLink
                className="tag-pill tag-default"
                to="/programming"
>
                programming
</TestLink>,
              <TestLink
                className="tag-pill tag-default"
                to="/reactjs"
>
                reactjs
</TestLink>,
            ],
            "className": "tag-list",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "programming",
              "nodeType": "function",
              "props": Object {
                "children": "programming",
                "className": "tag-pill tag-default",
                "to": "/programming",
              },
              "ref": null,
              "rendered": Object {
                "instance": <a
                  href="/programming"
                  class="tag-pill tag-default"
>
                  programming
</a>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "programming",
                  "className": "tag-pill tag-default",
                  "href": "/programming",
                },
                "ref": null,
                "rendered": Array [
                  "programming",
                ],
                "type": "a",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "reactjs",
              "nodeType": "function",
              "props": Object {
                "children": "reactjs",
                "className": "tag-pill tag-default",
                "to": "/reactjs",
              },
              "ref": null,
              "rendered": Object {
                "instance": <a
                  href="/reactjs"
                  class="tag-pill tag-default"
>
                  reactjs
</a>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "reactjs",
                  "className": "tag-pill tag-default",
                  "href": "/reactjs",
                },
                "ref": null,
                "rendered": Array [
                  "reactjs",
                ],
                "type": "a",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "tags": Array [
          Object {
            "link": "/programming",
            "name": "programming",
          },
          Object {
            "link": "/reactjs",
            "name": "reactjs",
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="sidebar"
>
          <p>
                    Popular Tags
          </p>
          <div
                    class="tag-list"
          >
                    <a
                              href="/programming"
                              class="tag-pill tag-default"
                    >
                              programming
                    </a>
                    <a
                              href="/reactjs"
                              class="tag-pill tag-default"
                    >
                              reactjs
                    </a>
          </div>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              Popular Tags
</p>,
            <div
              className="tag-list"
>
              <TestLink
                            className="tag-pill tag-default"
                            to="/programming"
              >
                            programming
              </TestLink>
              <TestLink
                            className="tag-pill tag-default"
                            to="/reactjs"
              >
                            reactjs
              </TestLink>
</div>,
          ],
          "className": "sidebar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <p>
              Popular Tags
</p>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Popular Tags",
            },
            "ref": null,
            "rendered": Array [
              "Popular Tags",
            ],
            "type": "p",
          },
          Object {
            "instance": <div
              class="tag-list"
>
              <a
                            href="/programming"
                            class="tag-pill tag-default"
              >
                            programming
              </a>
              <a
                            href="/reactjs"
                            class="tag-pill tag-default"
              >
                            reactjs
              </a>
</div>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <TestLink
                  className="tag-pill tag-default"
                  to="/programming"
>
                  programming
</TestLink>,
                <TestLink
                  className="tag-pill tag-default"
                  to="/reactjs"
>
                  reactjs
</TestLink>,
              ],
              "className": "tag-list",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "programming",
                "nodeType": "function",
                "props": Object {
                  "children": "programming",
                  "className": "tag-pill tag-default",
                  "to": "/programming",
                },
                "ref": null,
                "rendered": Object {
                  "instance": <a
                    href="/programming"
                    class="tag-pill tag-default"
>
                    programming
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "programming",
                    "className": "tag-pill tag-default",
                    "href": "/programming",
                  },
                  "ref": null,
                  "rendered": Array [
                    "programming",
                  ],
                  "type": "a",
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": "reactjs",
                "nodeType": "function",
                "props": Object {
                  "children": "reactjs",
                  "className": "tag-pill tag-default",
                  "to": "/reactjs",
                },
                "ref": null,
                "rendered": Object {
                  "instance": <a
                    href="/reactjs"
                    class="tag-pill tag-default"
>
                    reactjs
</a>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "reactjs",
                    "className": "tag-pill tag-default",
                    "href": "/reactjs",
                  },
                  "ref": null,
                  "rendered": Array [
                    "reactjs",
                  ],
                  "type": "a",
                },
                "type": [Function],
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
